src/route.ts

```diff
@@ -16,41 +16,51 @@ export class Route {
   get params() { return this.options.params; }
   get operationId() { return this.options.operationId; }
 
+  public getMiddlewareMetadata<MiddlewareMetadata>(middlewareClass: any) : MiddlewareMetadata | undefined {
```

Comment:
actually `middlewareClass` doesn't mean class (constructor),  it might be confused to newbies ;)

so it would be nice if we change this param name (e.g. `namespace: string`),

or change interface like this:

```typescript
abstract class Middleware {
  public static readonly METADATA_NAMESPACE: string;
  // ... some interfaces..
}

public getMiddlewareMetadata<MiddlewareMetadata>(middlewareClass: Middleware)  {
  return this.options.middlewareMetadata && this.options.middlewareMetadata[middlewareClass.METADATA_NAMESPACE];
} 
```

Reply:
originally i made it to be like CacheMiddleware: data, and do 
```
this.options.middlewareMetadata[middlewareClass.name]
```
i kinda wasn't sure this is really safe approach for all the ES6 / Typescript compilers. now i checked and it seems pretty safe, so i'll update it to just use "Class Name" of "Middleware Class"